File: Programmers/Lv1/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0

    for num in range(1, number + 1):
        cnt = 0  # 약수의 개수

        for i in range(1, int((num) ** 0.5) + 1):
            if num % i == 0:
                cnt += 1
                if i ** 2 != num:
                    cnt += 1

        if cnt > limit:
            answer += power
        else:
            answer += cnt

    return answer

File: Programmers/Lv1/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    for (let num = 1; num <=number; num ++ ){
        let cnt = 0
        for(let i = 1; i<=Math.sqrt(num);i++){
            if (num % i == 0){
                cnt += 1;
                if (num / i !== i){
                    cnt += 1;
                }
            }
            
        }
        if(cnt > limit){
            answer += power;
        } else {
            answer += cnt;
        }
    }
    
    return answer;
}

// 약수 구하기

var arr = [];
for(let idx = 1; idx <= Math.sqrt(num); idx ++){
    if(num % idx == 0){
        arr.push(idx);
        if(num / idx !== idx){
            arr.push(num / idx);
        }
    }
}

arr.sort((a, b) => a - b);